lib/Raisin/Entity.pm

```diff
@@ -69,6 +69,8 @@ sub compile {
         while (my $i = $data->next) {
             push @$result, _compile_column($entity, $i, \@expose);
         }
+
+        $result = [] unless $result;
     }
     # Array
     elsif (ref($data) eq 'ARRAY') {
```

Comment:
You might want to default `$result` to `[]` for the case where `ref($data) eq 'ARRAY'` as well.

Reply:
Isn't `$result` equals to `[]` if `ref($data) eq 'ARRAY'` returns true?

Reply:
It doesn't look like it. If we have `$data = []`, `$result` will remain `undef`. `$result` only gets a value inside the loop, on lines 76/78, below.

Reply:
Correct. Thanks for pointing out to that!